google_shopping.rb

```diff
@@ -8,8 +8,120 @@
+
+#Number 2, Find all items with a backorder availability in inventories.
+count_backorder = 0
+puts "Number 2 - prints all stock with 'backorder' availability **************************************************"
+puts "************************************************************************************************************"
+puts
+items.each do |items|
+    availability = items["product"]["inventories"][0]["availability"]
+    title = items["product"]["title"]
+    if availability.index("backorder")
+        puts " • " + title + ", has availability - " + availability 
+        count_backorder += 1
+    end
+end
+puts
+puts "There is/are " + count_backorder.to_s + " orders on backorder"
+puts
+puts "************************************************************************************************************"
+
+#Number 3, Find all items with more than one image link.
+puts "Number 3 - Display items with more than one image link *****************************************************"
+puts "************************************************************************************************************"
+puts
+count_image = 0
+items.each do |items|
+    image = items["product"]["images"]
+    title = items["product"]["title"]
+    if image.length > 1
+        puts " • " + title + ", has more than one image link."
+        count_image += 1
+    end
+end
+puts
+puts "there is/are " + count_image.to_s + " items with more than one image file"
+puts
+puts "************************************************************************************************************"
+
+#Number 4, Find all canon products in the items (careful with case sensitivity).
+puts "Number 4 - Display items by Canon **************************************************************************"
+puts "************************************************************************************************************"
+puts
+count_brand = 0
+items.each do |items|
+    title = items["product"]["title"]
+    brand = items["product"]["brand"]
+    if brand.downcase.include?("canon")
+        puts " • " + title + ", has brand " + brand
+        count_brand += 1
+    end
+end
+puts
+puts "There is/are " + count_brand.to_s + " Canon branded items"
+puts 
+puts "************************************************************************************************************"
+
+#Number 5, Find all items that have product author name of "eBay" and are brand "Canon".
+puts "Number 5 - Display items by author eBay and brand Canon ****************************************************"
+puts "************************************************************************************************************"
+puts
+count_authorbrand = 0
+items.each do |items|
+    author = items["product"]["author"]["name"]
+    brand = items["product"]["brand"]
+    title = items["product"]["title"]
+    if brand.downcase.include?("canon")
```

Comment:
You could also use && to avoid having to put in a nested if statement, like this:
```
if brand.downcase.include?("canon") && author.downcase.include?("ebay")
```